ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -420,6 +429,28 @@ LockState acquireLocks(QueryPlan plan, Context ctx, String username, boolean isB
     return lockState;
   }
 
+  private Collection<LockComponent> getGlobalLocks(Configuration conf) {
+    String lockNames = conf.get(ConfVars.HIVE_QUERY_EXCLUSIVE_LOCK.varname);
+    if (StringUtils.isEmpty(lockNames)) {
+      return Collections.emptyList();
+    }
+    List<LockComponent> globalLocks = new ArrayList<LockComponent>();
+    for (String lockName : lockNames.split(",")) {
```

Comment:
Can we end up having multiple locks here for schedule query? That should not be the case?